src/main/java/carpet/script/language/ControlFlow.java

```diff
@@ -145,5 +151,23 @@ public static void apply(Expression expression) // public just to get the javado
                 return (c_, t_) -> retval;
             }
         });
+
+        expression.addUnaryFunction("encode_b64", v -> StringValue.of(Base64.getEncoder().encodeToString(v.getString().getBytes())));
+        expression.addUnaryFunction("decode_b64", v -> {
+            try {
+                return StringValue.of(new String(Base64.getDecoder().decode(v.getString()), StandardCharsets.ISO_8859_1));//using this charset cos it's the one used in decoding function
+            } catch (IllegalArgumentException iae){
+                throw new InternalExpressionException("Invalid b64 string: " + v.getString());
```

Comment:
throw a user-catchable error rather here, there is already a json_error, just add base64_error

Reply:
wdym user catchable?

Reply:
ThrowStatement, from #692 + an edit I can't find the number rn. You have examples there.

(aka exceptions that can be caught by the user in a `try` block)

Reply:
oh ok